### PR TITLE
Support memref.collapse_shape, memref.expand_shape operation

### PR DESCRIPTION
**Pull request by makesource**

### CODE REVIEW

File: src/value.cpp [hunk 1]
```diff
@@ -1608,6 +1608,11 @@ MemRef MemRef::subview(const vector<Expr> &offsets,
   }
 }
 
+MemRef MemRef::reshape(const std::vector<smt::Expr> &newDims) {
```
Comment (aqjune) [inline src/value.cpp:1611]: Should we insert an assertion stating that this MemRef has an identity layout?

File: src/value.cpp [hunk 2]
```diff
@@ -1608,6 +1608,11 @@ MemRef MemRef::subview(const vector<Expr> &offsets,
...
+  return MemRef(m, elemType, bid, offset,
+    newDims, MemRef::Layout(newDims), Expr::mkBool(true));
```
Comment (aqjune) [inline src/value.cpp:1613]: Let's leave a comment in its declaration (value.h) saying that the returned MemRef is a view reference.

File: src/value.cpp (hunk 1 above)
Comment (makesource) [reply src/value.cpp:1611]: resolved

File: src/value.cpp (hunk 2 above)
Comment (makesource) [reply src/value.cpp:1616]: resolved

**Pull request merged**